src/main/java/orlov/programming/springcoregym/model/training/Training.java

```diff
@@ -1,22 +1,76 @@
 package orlov.programming.springcoregym.model.training;
 
+import jakarta.persistence.*;
 import lombok.Data;
+import lombok.Getter;
 import lombok.NoArgsConstructor;
+import lombok.Setter;
 import lombok.experimental.SuperBuilder;
+import org.hibernate.proxy.HibernateProxy;
 import orlov.programming.springcoregym.model.user.Trainee;
 import orlov.programming.springcoregym.model.user.Trainer;
 
 import java.time.LocalDate;
 import java.time.LocalTime;
+import java.util.Objects;
 
-@Data
+@Getter
+@Setter
 @NoArgsConstructor
 @SuperBuilder
+@Entity
 public class Training {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @ManyToOne
+    @JoinColumn(name = "trainee_id", nullable = false)
     private Trainee trainee;
+
+    @ManyToOne
+    @JoinColumn(name = "trainer_id", nullable = false)
     private Trainer trainer;
+
+    @Column(nullable = false)
     private String trainingName;
+
+    @ManyToOne
+    @JoinColumn(name = "training_type_id", nullable = false)
     private TrainingType trainingType;
+
+    @Column(nullable = false)
     private LocalDate trainingDate;
-    private LocalTime trainingDuration;
+
+    @Column(nullable = false)
+    private Long trainingDuration;
+
+    @Override
+    public String toString() {
```

Comment:
why not use Lombok to generate toString, equals, hashCode methods?

Reply:
Using Lombok with hibernate entity relationships results in many pitfalls and inconsistencies. If it's not a cornerstone, I would rather stick to the JPABuddy implementation than the lombok one.